Gemfile

```diff
@@ -28,31 +28,62 @@ group :system_tests do
   gem 'beaker-puppet_install_helper', :require => false
 end
 
-is_x64 = Gem::Platform.local.cpu == 'x64'
-if is_x64
-  platform(:x64_mingw) do
-    gem "win32-dir", "~> 0.4.9", :require => false
-    gem "win32-process", "~> 0.7.4", :require => false
-    gem "win32-service", "~> 0.8.4", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-  end
-else
-  platform(:mingw) do
-    gem "win32-process", "~> 0.6.5", :require => false
-    gem "win32-service", "~> 0.7.2", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-    if RUBY_VERSION =~ /^1\./ then
-      gem "win32console", :require => false
+
+ENV['GEM_PUPPET_VERSION'] ||= ENV['PUPPET_GEM_VERSION']
+puppetversion = ENV['GEM_PUPPET_VERSION']
+puppetversion = '~> 3' if !puppetversion
+gem 'puppet', *location_for(ENV['PUPPET_LOCATION'] || puppetversion)
+
+# For Windows dependencies, these could be required
+# based on the version of Puppet you are requiring.
+# Anything greater than v3.5.0 is going to have
+# Windows-specific dependencies dictated by the gem
+# itself.
+# The other scenario is when you are faking out the
+# Puppet Gem to use a local file path / git path
+explicitly_require_windows_gems = false
+if ENV['PUPPET_LOCATION'] || (ENV['GEM_PUPPET_VERSION'] && ENV['GEM_PUPPET_VERSION'] < '3.5.0')
+  is_x64 = Gem::Platform.local.cpu == 'x64'
+  if is_x64
+    platform(:x64_mingw) do
+      explicitly_require_windows_gems = true
+    end
+  else
+    platforms :mswin, :mingw do
```

Comment:
Kept around because we've had that on gemfiles in the past:
https://github.com/puppetlabs/puppetlabs-acl/blob/master/Gemfile#L39-L40